Aug 4 2021/Ch1ResnetV3.py

```diff
@@ -1,245 +1,225 @@
+             outputs = model(data) # run the images through the model
+             _,predicted = torch.max(outputs,1) # returns the index of the highest prediction
+             images_so_far +=1 # add one to the number of images used so far
+             image = data[i] # image = image tensor indexed at i
+             plt.imshow(image.cpu().squeeze(), cmap='gray') # plot the image
+             ground_truth = targets[i]
+             model_predicted = _[i]
+             title = ('ground truth',ground_truth,'model predicted',model_predicted) # set the title
+             plt.title(title) # add the title to the image
+             plt.show() # show the image
+             if images_so_far == num_images: # keep going over images until num_images = images_so_far
+                 return
 
-"""
-Step 9 Define Testing Logic 
-"""
 
-def validation(model, validation_loader, criterion):
-    valid_loss = 0
-    model.eval()
-    print('Validating model ')
-# Run the model on some test examples
-    with torch.no_grad():
-        correct, total = 0, 0
-        for data, targets in validation_loader:
-            data, targets = data.to(device), targets.to(device)
-            outputs = model(data)
-            loss = criterion(outputs, targets)
-            valid_loss += loss.item()
-            _, predicted = torch.max(outputs.data, 1)
-            total += targets.size(0)
-            correct += targets.sum().item()
-            # correct += (predicted == targets).sum().item()
-            # print(predicted)
-            # print(targets)
-
-        wandb.log({'Validation_average_loss': np.mean(valid_loss)})
-        print(f"Accuracy of the model on the{total}" + f"test images:{100*correct/total}%")
-        wandb.log({"test_accuracy": correct / total})
-
-
-
-# Visualizing model predictions
-
-def visualize_prediction(model,validation_loader,num_images=1):
-     # was_training = model.eval()
-     model.eval()
-     images_so_far = 0
-     fig = plt.figure()
-     with torch.no_grad():
-        for i,(data,targets) in enumerate(validation_loader):
-            data = data.to(device)
-            targets = targets.to(device)
-            outputs = model(data)
-            _,predicted = torch.max(outputs,1)
-            images_so_far +=1
-            plt.imshow(outputs.cpu(),cmap='gray')
-            plt.show()
-            if images_so_far == num_images:
-                return
-        # for j in range(data.size()[0]):
-        #     images_so_far += 1
-        #     plt.imshow(outputs.cpu(), cmap='gray')
-        #     plt.show()
-        #     plt.title('predicted:{}'.format(predicted[j]))
-        #     if images_so_far == num_images:
-        #         # model.train(model=was_training)
-        #         return
-
-    # model.train(mode=was_training)
 
 # Saving the model
 save_path = './mlp.pth'
-torch.save(model_pipeline(config).state_dict(), save_path)
+torch.save(model_pipeline(hyperparameter).state_dict(), save_path)
```

Comment:
I think this is fine, but you might want to separate these out into a couple lines if you see a bug. But I see that model_pipeline returns the trained model and then u grab the state_dict, so this looks good to me